api-gateway/src/main/resources/application.yml

```diff
@@ -117,7 +117,7 @@ spring:
           huawei:
             provider: huawei
             # hardcoded as for gitee
-            redirect-uri: 'https://saveourtool.com/login/oauth2/code/huawei'
+            redirect-uri: 'https://cosv.gitlink.org.cn/login/oauth2/code/huawei'
```

Comment:
what's about using `{baseUrl}`?

Reply:
Actually yes, very good point

Reply:
But stackoverflow actually says different:  https://stackoverflow.com/questions/66985696/how-does-default-redirect-url-work-in-spring-security-5

Reply:
If you use https://someserver:port/*, then it will match all endpoints on that server. When relying on the default Spring Security endpoint {baseUrl}/login/oauth2/code/{registrationId}, it's better to be specific in the redirect URL configuration and use https://someserver:port/login/oauth2/code/{registrationId} as the value. Both options work correctly, but the latter is more secure.

Reply:
But I will change to {baseUrl}